OpenRA.Game/Widgets/WorldInteractionControllerWidget.cs

```diff
@@ -236,10 +236,24 @@ public override bool HandleKeyPress(KeyInput e)
 				{
 					// Select actors on the screen which belong to the current player
 					var ownUnitsOnScreen = SelectActorsOnScreen(World, worldRenderer, null, player).SubsetWithHighestSelectionPriority();
```

Comment:
You'll want to stick a `ToList` on this (like `SelectUnitsByTypeKey` block does) to prevent it being enumerated from scratch multiple times.